### ISSUE TITLE
PyTorch 1.3 installation: No matching distribution found

### ISSUE THREAD

**Issue by haikuoY** [OpenMined/PySyft#2730]
ERROR: Could not find a version that satisfies the requirement torch==1.3 (from syft[udacity]) (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2)
ERROR: No matching distribution found for torch==1.3 (from syft[udacity])

**Comment by hasnain2808** [comment]
Hi
Are you using a virtual environment?
If yes then are you referring to the correct python installation

For example when you're in a virtual environment and want to install something via pip using this is better

python -m pip install [ package ]
Also are you on a supported python version?

**Comment by haikuoY** [comment]
my python version is 3.7.4, I think it's supported. 
I run the command python -m pip install syft[udacity] on cmd and I get the same error

**Comment by LaRiffle** [comment]
This looks like a pytorch installation error.
You should maybe ask on pytorch forums to try to understand how to install pytorch 1.3 with pip with you python / OS

**Comment by chunrongH** [comment]
Have you solved this issue?  I have met this problem today and I can not solve it.

**Comment by indrajitsg** [comment]
Even I am facing this same problem - any solution?

**Comment by indrajitsg** [comment]
I was finally able to solve the problem @chunrongH , @LaRiffle , @haikuoY , @hasnain2808 after a lot of trial and error. Here are the sequence of steps that I had followed in Windows 10.

1.  Installed Build Tools for Visual Studio 2019 from [here](https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019)
2. Created a new environment using conda `conda create --name syft python=3.7`
3. Installed the following packages from conda-forge:
```
conda install -c conda-forge websocket-client
conda install -c conda-forge websockets
conda install -c conda-forge lz4
conda install -c conda-forge tblib
conda install -c conda-forge msgpack-python
conda install -c conda-forge jupyterlab
conda install pytorch torchvision cpuonly -c pytorch
```
4. Went to proto repository [here](https://github.com/OpenMined/proto) and downloaded their repo and installed with:
```
python setup.py build
python setup.py install
```
5. Downloaded the syft repo [here](https://github.com/OpenMined/PySyft) and downloaded their repo and installed with:
```
python setup.py build
python setup.py install
```
The above process installed everything but ended with an error:
```
Searching for torch==1.3
Reading https://pypi.org/simple/torch/
No local packages or working download links found for torch==1.3
error: Could not find suitable distribution for Requirement.parse('torch==1.3')
```
6. Ignore the above error and start python. I was able to run `import syft as sy`.

**Comment by chunrongH** [comment]
Thank you for answering my question, I will try as you suggest.

At 2019-11-27 22:37:11, "Indrajit Sen Gupta" <notifications@github.com> wrote:


I was finally able to solve the problem @chunrongH , @LaRiffle , @haikuoY , @hasnain2808 after a lot of trial and error. Here are the sequence of steps that I had followed in Windows 10.

Installed Build Tools for Visual Studio 2019 from here
Created a new environment using conda conda create --name syft python=3.7
Installed the following packages from conda-forge:
conda install -c conda-forge websocket-client
conda install -c conda-forge websockets
conda install -c conda-forge lz4
conda install -c conda-forge tblib
conda install -c conda-forge msgpack-python
conda install -c conda-forge jupyterlab
conda install pytorch torchvision cpuonly -c pytorch

Went to proto repository here and downloaded their repo and installed with:
python setup.py build
python setup.py install

Downloaded the syft repo here and downloaded their repo and installed with:
python setup.py build
python setup.py install


The above process installed everything but ended with an error:

Searching for torch==1.3
Reading https://pypi.org/simple/torch/
No local packages or working download links found for torch==1.3
error: Could not find suitable distribution for Requirement.parse('torch==1.3')

Ignore the above error and start python. I was able to run import syft as sy.
